docs/cloud/dev/spryker-cloud-commerce-os/configuring-deployment-pipelines/deployment-in-states.md

```diff
@@ -0,0 +1,132 @@
+
+### Initial state
+This is how our application behaves  when no pipeline is running. In a fully working state:
+
+![Initial state](./images/initial_state/initial-state.gif)
+
+### Build_and_Prepare
+
+In this step, we build the Containers of each service that we are going to deploy (zed, yves, frontend, backoffice, backapi, glue, etc.)
+For the sake of simplicity, we will use only Glue and Zed in our examples. 
+
+![Build_and_Prepare](./images/Build_and_Prepare/Build_and_Prepare.jpg)
+
+### Configure_RabbitMQ_Vhosts_and_Permissions
+
+Here we update Rabbit MQ vhosts, users and permissions. 
+It should rarely be updated, but in this pessimistic approach, we want to illustrate what would happen if a request would come in while  RabbitMQ is updated:
+
+![Configure_RabbitMQ_Vhosts_and_Permissions](./images/Configure_RabbitMQ_Vhosts_and_Permissions/rmq.gif)
+
+### Run_pre-deploy_hook
+
+This step will run our application defined scripts. They are registered under the constant `SPRYKER_HOOK_BEFORE_DEPLOY` of our deploy.yml file. By default it  runs: 
+```shell
+vendor/bin/install -r pre-deploy -vvv
+```
+We mention this step here because it is important to understand that if our script takes a long time to run, we need to keep in mind all the services that are currently (or soon) in an updated state might respond incorrectly to requests. 
+
+An important part of this deployment step is this command:
+```shell
+vendor/bin/console scheduler:suspend -vvv --no-ansi
+```
+as it will stop our scheduler from running. Hereby preventing any data corruption or error for the duration of the deployment.
+It will first wait for the currently running jobs to finish and gracefully shutdown. (add this to the duration of the deployment)
+
+![scheduler:suspend](./images/Run_pre-deploy_hook/Run_pre-deploy_hook.jpg)
+
+Requests in the meantime are still coming in:
+
+![Run_pre-deploy_hook](./images/Run_pre-deploy_hook/pre_deploy.gif)
+
+### Deploy_Scheduler
+
+Now it's time to deploy the scheduler. It is based on the Zed container, as it needs to use Zed codebase. 
+
+![scheduler_paused](./images/Deploy_Scheduler/Deploy_Scheduler.jpg)
+
+Because the scheduler was paused, we don't need to fear it will run against incorrect data or services. 
+However, requests still are coming in:
+
+![Deploy_Scheduler](./images/Deploy_Scheduler/deploy_scheduler.gif)
+
+### Run_install
+
+This step should take care of running the necessary migrations for our new version (V2 here) of our application.
+The script that we run is associated with the constant: `SPRYKER_HOOK_INSTALL`.
+By default, this will run:
+```shell
+vendor/bin/install -r EU/production --no-ansi -vvv
+```
+
+After this step, the database should be updated. However, Search and Redis are not, as we "paused" the synchronization.
+
+![migrations](./images/Run_install/migrations.jpg)
+
+we should realize by now, that requests still coming in might error out. It all depends on what was migrated.
```

Comment:
Can you clarify this?

Reply:
the "depend on what was migrated" part

Reply:
Because the database has been migrated, v1 elements can run into issues because they don't know about the updated schemas. 

Recommend in conclusion: migrate as little as possible (divide updates into smaller chunks, so they take less downtime and produce less conflicts) and use feature flags whenever possible.